modules/core-bus-gghealthd/src/gg_healthd.c

```diff
@@ -3,44 +3,43 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "ggl/core_bus/gg_healthd.h"
+#include <ggl/arena.h>
 #include <ggl/buffer.h>
-#include <ggl/bump_alloc.h>
 #include <ggl/core_bus/client.h>
 #include <ggl/error.h>
 #include <ggl/log.h>
 #include <ggl/map.h>
 #include <ggl/object.h>
-#include <string.h>
 #include <stdint.h>
 
 GglError ggl_gghealthd_retrieve_component_status(
-    GglBuffer component, GglBuffer *component_status
+    GglBuffer component, GglArena *alloc, GglBuffer *component_status
 ) {
-    static uint8_t buffer[10 * sizeof(GglObject)] = { 0 };
-    GglBumpAlloc balloc = ggl_bump_alloc_init(GGL_BUF(buffer));
+    static uint8_t resp_mem[10 * sizeof(GglObject)] = { 0 };
+    GglArena resp_alloc = ggl_arena_init(GGL_BUF(resp_mem));
```

Comment:
Why not pass in the parameter from the function signature instead of creating a new allocator here?

Reply:
This memory should not consume space in callers allocator. The parameter is for allocating things returned to the caller. This stuff is used in this function and not returned to caller so should not be using caller's memory after returning.